akka-persistence-rs/src/effect.rs

```diff
@@ -552,6 +552,28 @@ where
 {
 }
 
+/// A side effect to run a function asynchronously and then, if ok,
+/// persist an event. The associated
+/// behavior is available so that communication channels, for
+/// example, can be accessed by the side-effect. Additionally, the
+/// latest state given any previous effect having persisted an event,
+/// or else the state at the outset of the effects being applied,
+/// is also available.
+pub fn persist_event_if<B, F, R>(f: F) -> ThenPersistEvent<B, F, R>
```

Comment:
is it possible to combine persist_event_if with a reply, so that you can reply with different things depending on the outcome of the async _if ? (I think so, but want to be sure)

Reply:
Yes, that’s possible. I’ve got an example of this which I’ll share when I’m back at my desk tomorrow.

Reply:
Here's my example:

```rust
let entity_id = context.entity_id.clone();
persist_event_if(move |b: &Self, _, _| {
    let registered = entity_id.parse().ok().map(|mac_address| {
        Event::Registered(Registered {
            client_ip: into_eui(&mac_address),
            client_pub_key,
            server_host: b.server_host.clone(),
            server_pub_key: b.server_pub_key.clone(),
        })
    });

    future::ready(Ok(registered))
})
.then_reply(move |s| {
    s.and_then(|s| {
        match (
            s.client_ip,
            &s.client_pub_key,
            &s.server_host,
            &s.server_pub_key,
        ) {
            (
                Some(client_ip),
                Some(client_pub_key),
                Some(server_host),
                Some(server_pub_key),
            ) => Some((
                reply_to,
                Registered {
                    client_ip,
                    client_pub_key: client_pub_key.clone(),
                    server_host: server_host.clone(),
                    server_pub_key: server_pub_key.clone(),
                },
            )),
            _ => None,
        }
    })
})
.boxed()
```

Reply:
thanks